model/sw_core.F90

```diff
@@ -1,3372 +1,3368 @@
-  integer, intent(in):: dir                ! 1: x-dir; 2: y-dir
-  real, intent(inout):: q(bd%isd:bd%ied,bd%jsd:bd%jed)
-  logical, intent(IN) :: sw_corner, se_corner, ne_corner, nw_corner
-  integer, intent(IN) :: npx, npy
-
-      integer :: is,  ie,  js,  je
-      integer :: isd, ied, jsd, jed
-
-      is  = bd%is
-      ie  = bd%ie
-      js  = bd%js
-      je  = bd%je
-      isd = bd%isd
-      ied = bd%ied
-      jsd = bd%jsd
-      jed = bd%jed
-
-  select case(dir)
-  case(1)
-      if ( sw_corner ) then
-          q(-1,0) = q(0,2)
-          q( 0,0) = q(0,1)
-      endif
-      if ( se_corner ) then
-          q(npx+1,0) = q(npx,2)
-          q(npx,  0) = q(npx,1)
-      endif
-      if ( nw_corner ) then
-          q( 0,npy) = q(0,npy-1)
-          q(-1,npy) = q(0,npy-2)
-      endif
-      if ( ne_corner ) then
-          q(npx,  npy) = q(npx,npy-1)
-          q(npx+1,npy) = q(npx,npy-2)
-      endif
-
-  case(2)
-      if ( sw_corner ) then
-          q(0, 0) = q(1,0)
-          q(0,-1) = q(2,0)
-      endif
-      if ( se_corner ) then
-          q(npx, 0) = q(npx-1,0)
-          q(npx,-1) = q(npx-2,0)
-      endif
-      if ( nw_corner ) then
-          q(0,npy  ) = q(1,npy)
-          q(0,npy+1) = q(2,npy)
-      endif
-      if ( ne_corner ) then
-          q(npx,npy  ) = q(npx-1,npy)
-          q(npx,npy+1) = q(npx-2,npy)
-      endif
-
-  end select
-
- end subroutine fill_4corners
-
- end module sw_core_mod
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the FV3 dynamical core.
+!*
+!* The FV3 dynamical core is free software: you can redistribute it
+!* and/or modify it under the terms of the
+!* GNU Lesser General Public License as published by the
+!* Free Software Foundation, either version 3 of the License, or
+!* (at your option) any later version.
+!*
+!* The FV3 dynamical core is distributed in the hope that it will be
+!* useful, but WITHOUT ANYWARRANTY; without even the implied warranty
+!* of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+!* See the GNU General Public License for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with the FV3 dynamical core.
+!* If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!>@brief The module 'sw_core' advances the forward step of the Lagrangian dynamics
+!! as described by \cite lin1997explicit, \cite lin2004vertically, and \cite harris2013two.
+!>@details The step is applied to the cubed sphere.
+
+ module sw_core_mod
+
+! Modules Included:
+! <table>
+! <tr>
+!    <th>Module Name</th>
+!     <th>Functions Included</th>
+!  </tr>
+!   <tr>
+!     <td>a2b_edge_mod</td>
+!     <td>a2b_ord4</td>
+!   <tr>
+!     <td>fv_arrays_mod</td>
+!     <td>ffv_grid_type, fv_grid_bounds_type, fv_flags_type</td>
+!   </tr>
```

Comment:
What's with all of the HTML at the beginning of each file?

Reply:
It's actually doxygen that lists the different module use statements in a table.  All of the different module use statements for FMS entries will go away within the next few months as we create a single libFMS module that will publicize everything necessary from FMS.  This will make FMS act like MPI and NetCDF when it comes to module inclusion.

Reply:
Thanks. So, doxygen puts this into the source file? Can we simply take out this HTML stuff?

Lucas

Reply:
Doxygen doesn't add anything automatically.  These were inserted when the code was doxygenated for EMC years ago.  With this synchronization of branches, differences between the dev/emc branch and the master/main branch of this nature will be ignored as we move forward.